src/gtc/dace/expansion/daceir_builder.py

```diff
@@ -263,9 +264,9 @@ def pop(self) -> "DaCeIRBuilder.IterationContext":
 
     @dataclass
     class SymbolCollector:
-        symbol_decls: Dict[SymbolRef, dcir.SymbolDecl] = dataclasses.field(default_factory=dict)
+        symbol_decls: Dict[str, dcir.SymbolDecl] = dataclasses.field(default_factory=dict)
```

Comment:
Is really `str` the correct typing? Why is not longer `SymbolRef`?

Reply:
There were some issues using `SymbolRef`. I can try adding it back in a separate PR, but for now let's stick with `str` and merge as soon as possible.

Reply:
Agree